tests/transport_test.go

```diff
@@ -48,7 +48,7 @@ var _ = Describe("Transport Tests", func() {
 
 	// it() is the body of the test and is executed once per Entry() by DescribeTable()
 	// closes over c and ns
-	it := func(ep string, credentialed bool) {
+	it := func(ep string, credentialed bool, accessKey, secretKey string, shouldSucceed bool) {
```

Comment:
`credentialed bool` should be dropped as it's redundant with `accessKey` and `secretKey`.

Reply:
I was thinking similar... if accessKey != "" || secretKey != "" { // then we have credentialed==true

Reply:
Maybe I'm missing something....
there are two scenarios we need to address:
1) no auth
2) with auth

In case of auth, we have two other options:
a) accessKey and secretKey taken from Utils
b) accessKey and secretKey passed as arguments.

The the example that you guys show above, it does not cover all options.
Jeff's example does not cover the case of Auth with keys from utils. In this case the keys we pass are empty but we need to take them from utils. We do not do that outside the if clause.
In other words, Jeff's example implement a case where at least one of the keys contains a value as an argument.

Bottom line, the logic I am trying to convey is that first we want to know if we ask for auth or not.
After we know that, in case we do need auth, we need to check whether the keys where passed or should we take the default (utils).

If we are not on the same track, we can talk over bj (Jon, since Jeff is in the west cost, we can do it just us as you and Jeff come with the same approach here).

Reply:
initially, I did not understand that you guys want me to pass the default keys in the second Entry call, so I let the code do this logic.  I change the code to reflect this now. Thanks!